### PR TITLE
lfortran: update to 0.28.0

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Update

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 14
Xcode 15

macOS 10A190
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by barracuda156** [comment]
@reneeotten Any idea what is going wrong with `py-yaml`?
```
Building dependency (1 of 1) 'py311-yaml' with variants '' ... [FAIL]
```

File: lang/lfortran/Portfile [hunk 1]
```diff
@@ -5,7 +5,7 @@ PortGroup               cmake 1.1
 PortGroup               compiler_blacklist_versions 1.0
 PortGroup               github 1.0
 
-github.setup            lfortran lfortran 0.23.0 v
```
Comment (reneeotten) [inline lang/lfortran/Portfile:8]: latest version is 0.28.0

Comment (reneeotten) [reply lang/lfortran/Portfile:8]: also, why can you not use the normal way of downloading the sources. `fetch.type git` is really not something that should be used... The description of why this is no needed is not very clear - did you try to convince upstream to fix their build system?

Comment (barracuda156) [reply lang/lfortran/Portfile:8]: @reneeotten Oh wow, they are fast. It was at 0.25.0 yesterday.

Let me check if normal way works now. Maybe the issue has been resolved, w/e it was.

**Comment by barracuda156** [comment]
@reneeotten Their script fails:
```
+ ci/version.sh
++ git describe --tags --dirty
fatal: not a git repository (or any of the parent directories): .git
+ version=
CMake Error at CMakeLists.txt:10 (message):
  Running build0.sh failed (see error above)


-- Configuring incomplete, errors occurred!
```
However, patching in the version manually seems to work fine.

File: lang/lfortran/Portfile (hunk 1 above)
Comment (barracuda156) [reply lang/lfortran/Portfile:8]: Switched to tarball.

**Comment by barracuda156** [comment]
@jmroot @mascguy @cjones051073 @reneeotten Well, okay, it is not a problem of R at all. CI are actually just broken again.

```
  Building 2 dependencies of lfortran:
  Building dependency (1 of 2) 'rapidjson' with variants '' ... [FAIL]
```

**Comment by barracuda156** [comment]
Amazing:
```
  From https://github.com/macports/macports-ports
   * branch              master     -> FETCH_HEAD
   * [new branch]        master     -> macports/master
  curl: (35) Recv failure: Connection reset by peer
  Error: Process completed with exit code 35.
```

**Comment by barracuda156** [comment]
@reneeotten Ok, this one at least is good.

**Comment by barracuda156** [comment]
@reneeotten @herbygillot This one should be good to merge.

**Pull request merged**